src/app/WriteClient.h

```diff
@@ -0,0 +1,141 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines read client for a CHIP Interaction Data model
+ *
+ */
+
+#pragma once
+
+#include <app/AttributePathParams.h>
+#include <app/InteractionModelDelegate.h>
+#include <app/MessageDef/AttributeDataList.h>
+#include <app/MessageDef/AttributeStatusElement.h>
+#include <app/MessageDef/WriteRequest.h>
+#include <core/CHIPCore.h>
+#include <core/CHIPTLVDebug.hpp>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace app {
+/**
+ *  @class WriteClient
+ *
+ *  @brief The read client represents the initiator side of a Write Interaction, and is responsible
+ *  for generating one Write Request for a particular set of attributes, and handling the Write response.
+ *
+ */
+class WriteClient : public Messaging::ExchangeDelegate
+{
+public:
+    /**
+     *  Shutdown the WriteClient. This terminates this instance
+     *  of the object and releases all held resources.
+     *
+     */
+
+    void Shutdown();
+
+    /**
+     * Finalize Write Request Message TLV Builder
+     *
```

Comment:
delete some of the empty  line.

The comment is a bit confusing: this  seems to finalize and take a packet, but the comment states something about the TLV builder. Please explain how this works (e.g. if it returns the final data from the tlv builder or something else)

Reply:
updated, thanks